### PR TITLE
Improved code of the app; Closes #1

### LINKED ISSUE #1: Improve the code with PR

### PR DESCRIPTION
**Pull request by juniwalk**
Příjde mi že mám málo zkušeností s refactoringem, mám pocit že jsem toho neudělal dost.

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Ahoj, díky za počkání. Jdu ti napsat feedback a oslovit tě s nabídkou práce na projektu https://costlocker.com. Na něm v rámci toho příkladu spolupracuji jako mentor.

Měl bys o něco takového zájem? :)

Jdu ti napsat feedback

File: app/bootstrap.php [hunk 1]
```diff
@@ -1,43 +1,41 @@
...
+use Dibi\Bridges\Nette\DibiExtension22;
 
-$configurator = new Nette\Configurator;
-$configurator->setTempDirectory(__DIR__ . '/runtime');
+include __DIR__.'/../vendor/autoload.php';
 
-// Enable Nette Debugger for error visualisation & logging
-$configurator->setDebugMode(DEBUG_MODE);
-$configurator->enableDebugger(__DIR__ . '/runtime/log');
+$di = new \Nette\Configurator;
```
Comment (TomasVotruba) [inline app/bootstrap.php]: I recommend keeping same name as class or what is the convention.
So `$configuration`.

File: app/bootstrap.php [hunk 2]
```diff
@@ -1,43 +1,41 @@
...
+$di->addConfig(__DIR__.'/config/config.neon');
+$di->setTempDirectory(__DIR__.'/../temp');
+$di->addParameters([
+	'pkgDir' => __DIR__.'/../vendor',
```
Comment (TomasVotruba) [inline app/bootstrap.php]: What is this for?

File: app/bootstrap.php [hunk 3]
```diff
@@ -1,43 +1,41 @@
...
+	$compiler->addExtension('dbDemoAccount', new DibiExtension22); // nazev sekce bude dbDemoAccount
 };
 
 DibiNettePanel::$maxLength = 2500;
 
-// Create Dependency Injection container from config.neon file
-$configurator->addConfig(__DIR__ . '/config/config.neon', APP_ENV);
-if (file_exists(__DIR__ . '/config/config.local.neon'))
-{
-    $configurator->addConfig(__DIR__ . '/config/config.local.neon', $configurator::NONE); // none section
```
Comment (TomasVotruba) [inline app/bootstrap.php]: Why remove `config.local.neon`? It's very useful

File: app/bootstrap.php [hunk 4]
```diff
@@ -1,43 +1,41 @@
...
-}
+$di->createRobotLoader()
+	->addDirectory(__DIR__)
```
Comment (TomasVotruba) [inline app/bootstrap.php]: Good point, robot loader is missing!

You can replace it with PSR4 `autoload` in  `composer.json` nowadays.

File: app/bootstrap.php [hunk 5]
```diff
@@ -1,43 +1,41 @@
...
 // SECURE COOKIES on HTTPS
-if($container->getService('httpRequest')->isSecured()) {
-
-    $container->getService('httpResponse')->cookieSecure = TRUE;
-
-    $container->getService('session')->setOptions(array('cookie_secure' => TRUE));
-
+if ($container->getService('httpRequest')->isSecured()) {
+	$container->getService('httpResponse')->cookieSecure = TRUE;
+	$container->getService('session')->setOptions(['cookie_secure' => TRUE]);
```
Comment (TomasVotruba) [inline app/bootstrap.php]: New array, I like it!

This should be configured in `config.neon` or in standalone extension.

Programming in `bootstrap.php` is not very reusable, nor common place to change stuffs.

File: app/model/BatchReplace.php [hunk 6]
```diff
@@ -4,57 +4,74 @@
...
-            if(isset($this->keyReplacements[$value])) {
-                $value = $this->keyReplacements[$value];
-            }
-        }
-
-        $this->mapper->replace($this->data, $this->field);
-
-        return array();
+	/** @var Mapper */
+	private $mapper;
```
Comment (TomasVotruba) [inline app/model/BatchReplace.php]: Try keeping the same code style in which you found the code.

Tabs/spaces can make a lot of mess

File: app/presenters/AbstractPresenter.php [hunk 7]
```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Presenters;
+
+use Model\ClAccountInfo;
+
+abstract class AbstractPresenter extends extends \Nette\Application\UI\Presenter
```
Comment (TomasVotruba) [inline app/presenters/AbstractPresenter.php]: Creating abstract presenter makes only sense while gathering repeated code from multiple presenters.

Here it has no added value.

Comment (TomasVotruba) [inline app/presenters/AbstractPresenter.php]: Also watch out for double `extends extends` here.

Did you use PHPStorm or some other IDE?

File: app/www/index.php [hunk 8]
```diff
@@ -1,19 +1,10 @@
...
- */
-$container = require dirname(__DIR__) . '/bootstrap.php';
-
-/**
- * Run application.
- * @var \Nette\Application\Application $application
- */
-$application = $container->getByType('Nette\Application\Application');
-$application->run();
+$di = include __DIR__.'/../boottrap.php';
```
Comment (TomasVotruba) [inline app/www/index.php]: Typo

File: app/www/index.php [hunk 9]
```diff
@@ -1,19 +1,10 @@
...
- * @var \Nette\DI\Container $container
- */
-$container = require dirname(__DIR__) . '/bootstrap.php';
-
-/**
- * Run application.
- * @var \Nette\Application\Application $application
- */
-$application = $container->getByType('Nette\Application\Application');
-$application->run();
```
Comment (TomasVotruba) [inline app/www/index.php]: Nice refactoring to less code!

File: app/www/index.php [hunk 10]
```diff
@@ -1,19 +1,10 @@
...
+$di = include __DIR__.'/../boottrap.php';
+$di->getByType('Nette\Application\Application')->run();
```
Comment (TomasVotruba) [inline app/www/index.php]: Here you could also use PHP 5.5 `::class` feature

```php
$container->getByType(Nette\Application\Application::class)->run();
```

File: app/bootstrap.php (hunk 2 above)
Comment (juniwalk) [inline app/bootstrap.php]: You mean those parameters? It's little trick I learned from Filip Procházka, it allows console comands to work properly.

File: app/bootstrap.php [hunk 11]
```diff
@@ -1,43 +1,41 @@
...
+$di = new \Nette\Configurator;
+$di->addConfig(__DIR__.'/config/config.neon');
+$di->setTempDirectory(__DIR__.'/../temp');
+$di->addParameters([
+	'pkgDir' => __DIR__.'/../vendor',
+	'wwwDir' => __DIR__.'/../www',
+	'appDir' => __DIR__,
+]);
+
+$di->setDebugMode(include __DIR__.'/config/config-ipconf.php');
```
Comment (juniwalk) [inline app/bootstrap.php]: That should've not been there, that is IP based debug mode which is excluded from git. It returns simple array with IPs.

File: app/bootstrap.php [hunk 12]
```diff
@@ -1,43 +1,41 @@
...
 mb_internal_encoding('UTF-8');
 
-// \Tracy\Debugger::$maxDepth = 5;
-
-$configurator->onCompile[] = function($configurator, $compiler) {
-    $compiler->addExtension('dibiPayments', new \Dibi\Bridges\Nette\DibiExtension22());
-    $compiler->addExtension('dbDemoAccount', new \Dibi\Bridges\Nette\DibiExtension22()); // nazev sekce bude dbDemoAccount
+$configurator->onCompile[] = function ($configurator, $compiler) {
+	$compiler->addExtension('dibiPayments', new DibiExtension22);
+	$compiler->addExtension('dbDemoAccount', new DibiExtension22); // nazev sekce bude dbDemoAccount
```
Comment (juniwalk) [inline app/bootstrap.php]: Yeah I know, I do this automaticaly, you said only to modify existing files so I left it alone. I would of course normaly use `config.neon`.

File: app/bootstrap.php (hunk 3 above)
Comment (juniwalk) [inline app/bootstrap.php]: I am used to `config-params.neon` included directly in `config.neon`, just my taste...

File: app/bootstrap.php (hunk 4 above)
Comment (juniwalk) [inline app/bootstrap.php]: I tried it, had some issues with it so I reverted back to robot loader for now.

File: app/www/index.php (hunk 10 above)
Comment (juniwalk) [inline app/www/index.php]: Yeah but it is longer, I usualy use `::class` when I have `use` -d the class in the php file so I don't rewrite whole namespace.

File: app/www/index.php (hunk 8 above)
Comment (juniwalk) [inline app/www/index.php]: Dammit, looks like I've rushed it too much.

File: app/presenters/AbstractPresenter.php (hunk 7 above)
Comment (juniwalk) [inline app/presenters/AbstractPresenter.php]: Well check the methods of the `AbstractPresenter`, there is Head components, which should probably be used in more presenters, also in `startup` method there is some tracking and XSRF  token being set, isn't that meant to be on all presenters?

Comment (juniwalk) [inline app/presenters/AbstractPresenter.php]: I used Rapid PHP, I did not get used to PHPStorm.

**Comment by juniwalk** [comment]
@TomasVotruba Co bude práce na tom projektu všechno obnášet? Máš nějakou představu? Jinak promiň jestli jsi chtěl komunikaci v angličtině, do budoucna samozřejmě není problém :)

File: app/bootstrap.php (hunk 2 above)
Comment (TomasVotruba) [inline app/bootstrap.php]: But what does it fix/improve in this code's scope.

File: app/bootstrap.php (hunk 11 above)
Comment (TomasVotruba) [inline app/bootstrap.php]: And where is the file?

File: app/bootstrap.php (hunk 3 above)
Comment (TomasVotruba) [inline app/bootstrap.php]: And how do you differ between local and production environment?

File: app/bootstrap.php (hunk 4 above)
Comment (TomasVotruba) [inline app/bootstrap.php]: What issues exactly? I might help with that.

File: app/presenters/AbstractPresenter.php (hunk 7 above)
Comment (TomasVotruba) [inline app/presenters/AbstractPresenter.php]: > @juniwalk Well check the methods of the AbstractPresenter, there is Head components, which should probably be used in more presenters, also in startup method there is some tracking and XSRF token being set, isn't that meant to be on all presenters?

This was ambiguous, I agree your assumptions are valid.

Comment (TomasVotruba) [inline app/presenters/AbstractPresenter.php]: @juniwalk Too bad. PHPStorm is great, I hated it first 2 years I knew it, but then I checked https://laracasts.com/series/how-to-be-awesome-in-phpstorm and it gave me another sight on it.

Also, there has been mayor evolution in last year or so. When did you last tried it?

File: app/www/index.php (hunk 10 above)
Comment (TomasVotruba) [inline app/www/index.php]: Length is not the issue, still you can use `use` statement in the head of file.

- This is very helpful while refactoring.
- Also it prevents typos.
- Autocomplete is much faster then typing manually.

**Comment by TomasVotruba** [comment]
@juniwalk Ze začátku hlavně refaktoring do čitelné podoby. S tím souvisí i to, že v kódu používáš hodně nestandartní a custom věci bez vysvětlení, takže to by bylo potřeba napravit. Kód projektu vypadá v tomhle podobně a to ho hodně znepřehledňuje.

Jsi z Prahy?

**Comment by juniwalk** [comment]
@TomasVotruba Ne ne, žiju v Liberci. Já většinou docela často koukám na to jak problémy řeší ostatní a pak si z toho něco osvojím, popřípadě spojím víc řešení. Většinou podle toho co lépe chápu a vím že není bad practice.

**Comment by TomasVotruba** [comment]
@juniwalk Tipnu si: děláš v týmu?

File: app/bootstrap.php (hunk 2 above)
Comment (juniwalk) [inline app/bootstrap.php]: Nothing, you're right, I just used my bootstrap.php as I am used to it.

File: app/bootstrap.php (hunk 11 above)
Comment (juniwalk) [inline app/bootstrap.php]: You said to modify only existing files so I did not add it.

File: app/bootstrap.php (hunk 3 above)
Comment (juniwalk) [inline app/bootstrap.php]: I don't differ, each enviroment has it's own copy of the file. The file itself is outside vcs.

File: app/presenters/AbstractPresenter.php (hunk 7 above)
Comment (juniwalk) [inline app/presenters/AbstractPresenter.php]: @TomasVotruba Well it was about a year ago. I may give it another try then.

File: app/www/index.php (hunk 10 above)
Comment (juniwalk) [inline app/www/index.php]: @TomasVotruba Yeah, you made good points.

**Comment by juniwalk** [comment]
@TomasVotruba Ano i ne, víceméně si každej hraje na svém písečku.

File: app/bootstrap.php (hunk 4 above)
Comment (juniwalk) [inline app/bootstrap.php]: @TomasVotruba With factories, I am used to write factory interfaces below the classes, which of course is not PSR-4 compatible.

File: app/presenters/AbstractPresenter.php (hunk 7 above)
Comment (TomasVotruba) [inline app/presenters/AbstractPresenter.php]: @juniwalk It's quite mainstream and with fast development.
Also you can win a license every month in http://pehapkari.cz community or you can use it for free with EAP - in the end of page http://pehapkari.cz/pro-budouci-programatorky/

**Comment by TomasVotruba** [comment]
Právě hodně věcí co v tom PR píšu se většinou učí v rámci týmové spolupráce. 

Dřív jsem si taky psal custom věci, které mi fungovali a vyhovovali. Až když jsem musel pracovat s 3 dalšíma lidma, kdy jsme se potřebovali domluvit a psát kód co nejsrozumitelněji to jde, tak jsem se učil tyhle návyky.

Doporučuju práci v týmu, on-site, klidně na měsíc. Neskutečně mě to posunulo, bo jsem si začal uvědomovat, jak je můj kód vnímán a jak já vnímám jiný kód.

Na tuhle pozici bych potřeboval někoho, kdo má tohle za sebou. Je to pracné to učit, když není tým zaběhlý, což tady není.

**Comment by juniwalk** [comment]
@TomasVotruba Rozumím, on-site se mi asi jen tak nepoštěstí ale teď mám nával nových zkušeností díky spolupráci na ublaboo/datagrid s Pavlem Jandou.

**Comment by TomasVotruba** [comment]
@juniwalk Paráda, Pavel je mistr :)

**Pull request closed without merging**